src/main/java/com/internet/shop/dao/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,187 @@
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        ShoppingCart shoppingCart = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT id_shoping_cart FROM shoping_cart "
+                    + " WHERE id_user = ? AND deleted = FALSE";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, userId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = new ShoppingCart();
+                shoppingCart.setId(resultSet.getLong("id_shoping_cart"));
+                shoppingCart.setUserId(userId);
+                shoppingCart.setProducts(getCartProducts(shoppingCart.getId()));
+                return Optional.of(shoppingCart);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Shopping cart by user id" + userId
+                    + " was not found", e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "INSERT INTO shoping_cart (id_user) VALUES (?)";
+            PreparedStatement preparedStatement
+                    = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setLong(1, shoppingCart.getUserId());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                shoppingCart.setId(resultSet.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Shopping cart with id "
+                    + shoppingCart.getId() + " was not created", e);
+        }
+        return shoppingCart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        ShoppingCart shoppingCart = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM shoping_cart "
+                    + "WHERE id_shoping_cart = ? AND deleted = FALSE";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = new ShoppingCart();
+                shoppingCart.setId(id);
+                shoppingCart.setUserId(resultSet.getLong("id_user"));
+                shoppingCart.setProducts(getCartProducts(shoppingCart.getId()));
```

Comment:
same with two connections at the same time

Reply:
fixed